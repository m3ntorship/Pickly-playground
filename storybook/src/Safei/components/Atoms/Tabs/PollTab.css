/* .tab-poll{
  @apply group text-sm relative flex items-center justify-around px-3.5 py-1.5 h-14 border border-grey border-opacity-0 bg-grey-shd7  hover:bg-white hover:border-opacity-100  rounded-full focus:outline-none transition duration-500 ease-in-out;
}
.tab-poll{
  ;
}
.checkmark{
  @apply h-4 w-4 rounded-full border border-primary  bg-white text-transparent absolute left-0 top-1 mr-2 hover:bg-grey-shd4 transition duration-500 ease-in-out;
} */

input:checked ~ .checkmark {
  background-color: #7048E8;
}

.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}
/* Show the indicator (dot/circle) when checked */
input:checked ~ .checkmark:after {
  display: block;
}
/* Style the indicator (dot/circle) */
.checkmark:after {
	top: 4px;
	left: 4px;
	width: 6px;
	height: 6px;
	border-radius: 999px;
	background: white;
	-webkit-border-radius: 999px;
	-moz-border-radius: 999px;
	-ms-border-radius: 999px;
	-o-border-radius: 999px;
}
