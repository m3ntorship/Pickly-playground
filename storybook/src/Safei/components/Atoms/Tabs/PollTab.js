import React from "react";
import "./PollTab.css";

const PollTab = () => {
  return (
    <button
      className="group text-sm relative flex items-center justify-around px-3.5 py-1.5 h-8 border border-grey border-opacity-0 bg-grey-shd7  hover:bg-white hover:border-opacity-100  rounded-full focus:outline-none transition duration-500 ease-in-out"
      style={{ paddingRight: "12px", marginRight: "18px" }}
    >
      <label className="cursor-pointer relative flex items-center justify-items-center content-center">
        <input type="radio" name="radio" className="bg-primary" />
        <span className="checkmark h-4 w-4 rounded-full border border-primary  bg-white text-transparent absolute left-0 top-1 mr-2 hover:bg-grey-shd4 transition duration-500 ease-in-out">
          {" "}
        </span>
        <svg
          className="w-6 h-6 ml-2"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="M4 16l4.586-4.586a2 2 0 012.828 0L16 16m-2-2l1.586-1.586a2 2 0 012.828 0L20 14m-6-6h.01M6 20h12a2 2 0 002-2V6a2 2 0 00-2-2H6a2 2 0 00-2 2v12a2 2 0 002 2z"
          ></path>
        </svg>
        <span className="ml-2">Image Poll</span>
      </label>
    </button>
  );
};

export default PollTab;
