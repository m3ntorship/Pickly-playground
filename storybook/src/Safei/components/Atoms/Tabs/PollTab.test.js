import React from "react";
import renderer from "react-test-renderer";
import PollTab from "./PollTab";

describe("Test my PollTab components", () => {
  //(1)
  it("renders a PollTab Button ", () => {
    const tree = renderer.create(<PollTab />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
