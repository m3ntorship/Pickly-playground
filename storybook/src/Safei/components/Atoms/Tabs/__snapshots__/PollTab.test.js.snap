// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my PollTab components renders a PollTab Button  1`] = `
<button
  className="group text-sm relative flex items-center justify-around px-3.5 py-1.5 h-8 border border-grey border-opacity-0 bg-grey-shd7  hover:bg-white hover:border-opacity-100  rounded-full focus:outline-none transition duration-500 ease-in-out"
  style={
    Object {
      "paddingRight": "12px",
    }
  }
>
  <label
    className="cursor-pointer relative flex items-center justify-items-center content-center"
  >
    <input
      className="bg-primary"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark h-4 w-4 rounded-full border border-primary  bg-white text-transparent absolute left-0 top-1 mr-2 hover:bg-grey-shd4 transition duration-500 ease-in-out"
    >
       
    </span>
    <svg
      className="w-6 h-6 ml-2"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 16l4.586-4.586a2 2 0 012.828 0L16 16m-2-2l1.586-1.586a2 2 0 012.828 0L20 14m-6-6h.01M6 20h12a2 2 0 002-2V6a2 2 0 00-2-2H6a2 2 0 00-2 2v12a2 2 0 002 2z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <span
      className="ml-2"
    >
      Image Poll
    </span>
  </label>
</button>
`;
